### ISSUE TITLE
Haul support for other platforms

### ISSUE THREAD

**Issue by rdy** [callstack/haul#385]
I'm looking to use haul to add support for other platforms.

Would there be any objection to looking in the project root for the rn-cli.config.js which metro uses to allow a place to specify the platforms.

For example:
`rn-cli.config.js`
```
module.exports = {
  getPlatforms: () => ['my-platform'],
}
```

We could read the file to grab the platforms out as well as add additional modules using haste.
```
 getProvidesModuleNodeModules: () => ['react-native', 'react-native-my-platform']
```

This would allow haul to be used in a similar way to metro in supporting other react-native targets without having to push the configuration somewhere directly in haul.

I can send a PR that demonstrates the idea, I have a working version of it already with Haul. The nice thing about this approach is that we wouldn't have to modify haul's source (by forking and adding platforms). Instead all the information can come from the same place that metro expects it. This would make it easier for folks coming from metro who have already extended `rn-cli.config.js.`

Thoughts?

**Comment by thymikee** [comment]
Cc @zamotany

**Comment by zamotany** [comment]
@rdy Can you make a repository demonstrating this idea??

**Comment by rdy** [comment]
sure thing, happy to do so. i was just checking to see if the approach seemed sound. i'll put something together.

**Comment by rdy** [comment]
I sent a PR #388 to demonstrate the idea, I'm happy to make adjustments to make it work. I based it against on the commit right before webpack 4 just because in my test application I can't migrate it over safely.

There are a couple of interesting changes, I had to suppress the duplicate haste module error, and instead I just took the first one, but other than that it is pretty straightforward.

If you want to see a repo that uses the module I send you that but all it needs is a metro style `rn-cli.config.json`
